src/dataspec/base.py

```diff
@@ -150,77 +150,144 @@ def tag(self) -> Tag:  # pragma: no cover
+    def conform_valid(self, v: Any):
         """
         Conform ``v`` to the Spec without checking if v is valid first and return the
         possibly conformed value or ``INVALID`` if the value cannot be conformed.
 
         This function should be used only if ``v`` has already been check for validity.
-        """
-        return self.conformer(v)
 
-    def conform(self, v: Any):
-        """Conform ``v`` to the Spec, first checking if v is valid, and return the
-        possibly conformed value or ``INVALID`` if the value cannot be conformed."""
-        if self.is_valid(v):
-            return self.conform_valid(v)
-        else:
+        :param v: a *validated* value to conform
+        :return: a conformed value or a :py:class:`dataspec.Invalid` instance if the
+            input value could not be conformed
+        """
+        if self.conformer is None:
+            return v
+        try:
+            return self.conformer(v)  # pylint: disable=not-callable
+        except Exception:
             return INVALID
 
-    def with_conformer(self, conformer: Conformer) -> "Spec":
-        """Return a new Spec instance with the new conformer."""
+    def compose_conformer(self, conformer: Conformer) -> "Spec":
+        """
+        Return a new Spec instance with a new conformer which is the composition of the
+        ``conformer`` and the current conformer for this Spec instance.
+
+        If the current Spec instance has a custom conformer, this is equivalent to
+        calling ``spec.with_conformer(lambda v: new_conformer(spec.conformer(v)))`` .
+        If the current Spec instance has no custom conformer, this is equivalent to
+        calling :py:meth:`dataspec.Spec.with_conformer` with ``conformer`` .
+
+        To completely replace the conformer for this Spec instance, use
+        :py:meth:`dataspec.Spec.with_conformer` .
+
+        This method does not modify the current Spec instance.
+
+        :param conformer: a conformer to compose with the conformer of the current
+            Spec instance
+        :return: a copy of the current Spec instance with the new composed conformer
+        """
+        existing_conformer = self.conformer
+
+        if existing_conformer is None:
+            return self.with_conformer(conformer)
+
+        def conform_spec(v: T) -> Union[V, Invalid]:
+            assert existing_conformer is not None
+            return conformer(existing_conformer(v))  # pylint: disable=not-callable
+
+        return self.with_conformer(conform_spec)
+
+    def with_conformer(self, conformer: Optional[Conformer]) -> "Spec":
+        """
+        Return a new Spec instance with the new conformer, replacing any custom
+        conformers.
+
+        If ``conformer`` is :py:obj:`None` , the returned Spec will have no custom
+        conformer.
+
+        To return a copy of the current Spec with a composition of the current
+        Spec instance, use :py:meth:`dataspec.Spec.compose_conformer` .
```

Comment:
```suggestion
        Spec instance, use :py:meth:`dataspec.Spec.compose_conformer`.
```

Reply:
I've been adding these spaces after object references because Sphinx was complaining about having trailing text after a cross-reference. Here's an example where I removed the space between `dataspec.Conformer` and `s` in the docstring for `class Invalid`:

```
/Users/chrisrink/Projects/dataspec/src/dataspec/base.py:docstring of dataspec.Invalid:1: WARNING: Inline interpreted text or phrase reference start-string without end-string.
```

Curiously, I was able to fix all of the instances in the current docs and docstrings with trailing _periods_ and the docs are building now. I'm not exactly clear what's happening 🤷‍♂️